### DISCUSSION TITLE
createHref support.

### DISCUSSION THREAD

**Discussion by hellohejinyu** [remix-run/react-router#9842] [Proposals]
### What is the new or updated feature that you are suggesting?

`createHref` need back.In some scenarios, you need to dynamically calculate the url and then use it.

### Why should this feature be included?

In the previous v5 version, I can get the href in the following code.
```jsx
import { useHistory } from 'react-router-dom'

const Cpt = () => {
  const history = useHistory()

  return <button 
    onClick={() => {
      const href = history.createHref('/bar')
      window.open(`http://some-other-domain.com${href}`)
    }}
  >
    button
  </button>
}
```

But in the latest version, I can't achieve my needs.I just found the `resolvePath` method.But its result is inconsistent with `createHref`. And `useHref` cannot dynamically pass in parameters.

![image](https://user-images.githubusercontent.com/8766034/140896920-f39bcb1e-a618-423a-a325-84e88c5cea79.png)

So, I think `createHref` is still necessary for its existence, or I missed something, there is a better way to achieve it in the v6 version.

**Comment by timdorr** [comment]
Related to #8278

**Comment by brophdawg11** [comment]
I'm going to convert this to a discussion so it can go through our new [Open Development](https://remix.run/blog/open-development) process.  Please upvote the new Proposal if you'd like to see this considered!

**Reply by brophdawg11** [reply]
If we decide to move forward with this proposal, there's an implementation in this closed PR: https://github.com/remix-run/react-router/pull/9161
